crypto/crypto.go

```diff
@@ -70,8 +62,14 @@ func Decrypt(ciphertext []byte, key []byte) ([]byte, error) {
 }
 
 // DeriveKey derives a 32 byte encryption key from a password and identifier.
-func DeriveKey(password, identifier []byte, cost map[string]int) []byte {
-	derivedKey, _ := scrypt.Key(password, identifier, 1<<uint(cost["N"]), cost["r"], cost["p"], 32)
+func DeriveKey(password, identifier []byte, cost map[string]int) [32]byte {
+	//LOCKTHIS
+	derivedKeySlice, _ := scrypt.Key(password, identifier, 1<<uint(cost["N"]), cost["r"], cost["p"], 32)
+
+	// Convert to fixed-size array.
```

Comment:
Not everyone reading the source code would necessarily know that arrays are fixed size. I think it's better to leave that in there instead of assuming the reader's previous knowledge.